### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by puiterwijk**

### CODE REVIEW

File: fmn/consumer/backend.py [hunk 1]
```diff
@@ -231,10 +231,10 @@ def read(queue_object):
...
 lc3.start(frequency)
 
-digest_producer = fmn_producers.DigestProducer(
-    session, backends)
-lc4 = task.LoopingCall(digest_producer.work)
-lc4.start(frequency)
+#digest_producer = fmn_producers.DigestProducer(
+#    session, backends)
+#lc4 = task.LoopingCall(digest_producer.work)
+#lc4.start(frequency)
```
Comment (bowlofeggs) [inline fmn/consumer/backend.py]: If you want to drop this code, let's just delete it - if we need it back one day it'll be in the git history.

File: fmn/consumer/backends/mail.py [hunk 2]
```diff
@@ -151,9 +151,11 @@ def _format_line(msg):
         n = len(queued_messages)
         subject = u"Fedora Notifications Digest (%i updates)" % n
         summary = u"Digest summary:\n"
+        i = 0
         for msg in queued_messages:
+            i = i + 1
             line = fedmsg.meta.msg2subtitle(msg, **self.config) or u''
-            summary += str(i+1) + ".\t" + line + "\n"
+            summary += str(i) + ".\t" + line + "\n"
```
Comment (bowlofeggs) [inline fmn/consumer/backends/mail.py]: Instead of the above, how about:

```
for i, msg in queued_messages:
    …
    summary += "{}.\t{}\n".format(str(i + 1), line)
```

Or you can use the %s format, since I know you like that ☺ `enumerate` is a nicer way to generate indices in for loops though, and you can drop two lines of code!

File: fmn/consumer/backends/mail.py [hunk 3]
```diff
@@ -176,8 +178,19 @@ def _format_line(msg):
             fedmsg.meta.msg2packages(msg, **self.config)
             for msg in queued_messages])
 
-        self.send_mail(session, recipient, subject, content,
-                       topics, categories, usernames, packages)
+        logger.info('Breaking email')
+        logger.info('Length: %d' % len(content))
+
+        # Break email
+        contents = []
```
Comment (bowlofeggs) [inline fmn/consumer/backends/mail.py]: You don't need this line since contents gets set again two lines below.

File: fmn/consumer/backends/mail.py [hunk 4]
```diff
@@ -176,8 +178,19 @@ def _format_line(msg):
...
+        split = 20000000
+        contents = [content[i:i+split] for i in range(0, len(content), split)]
+
+        logger.info('Broken email into %d parts' % len(contents))
```
Comment (bowlofeggs) [inline fmn/consumer/backends/mail.py:188]: Maybe you want an `if len(contents) > 1` over this so it doesn't log "normal" e-mails?

File: fmn/consumer/producer.py [hunk 5]
```diff
@@ -55,7 +55,7 @@ def work(self):
         for pref in fmn.lib.models.Preference.list_batching(self.session):
 
             if not pref.context.name in self.backends:
-                return
+                continue
```
Comment (bowlofeggs) [inline fmn/consumer/producer.py:59]: This seems like a pretty significant change in behaviour. Why is this needed?

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
Can you drop the commented code and answer my "why is this needed?" question before we merge?

Everything else is just a suggestion, so you can ignore them if you like.

File: fmn/consumer/producer.py (hunk 5 above)
Comment (puiterwijk) [reply fmn/consumer/producer.py:59]: Because with the current code it silently crashes (stops), because it doesn't ack the next message.
This is the case where we are unable to send a specific type of mesasge, so let's just ignore it for now.

Comment (bowlofeggs) [reply fmn/consumer/producer.py:59]: Oh, and can you answer the "why is this needed" by adding a comment to the code? ☺

Comment (puiterwijk) [reply fmn/consumer/producer.py:59]: Well, this just makes it in line with the others below it...

Comment (puiterwijk) [reply fmn/consumer/producer.py:59]: Added a log entry.

File: fmn/consumer/backends/mail.py (hunk 4 above)
Comment (puiterwijk) [reply fmn/consumer/backends/mail.py:188]: Nah, why?

File: fmn/consumer/backends/mail.py (hunk 3 above)
Comment (puiterwijk) [reply fmn/consumer/backends/mail.py]: Removed

File: fmn/consumer/backends/mail.py (hunk 2 above)
Comment (puiterwijk) [reply fmn/consumer/backends/mail.py]: Fixed.

File: fmn/consumer/backends/mail.py (hunk 4 above)
Comment (bowlofeggs) [reply fmn/consumer/backends/mail.py:188]: Just so you keep the log noise down - it's just a suggestion ☺

File: fmn/consumer/backends/mail.py [hunk 6]
```diff
@@ -151,9 +151,11 @@ def _format_line(msg):
         n = len(queued_messages)
         subject = u"Fedora Notifications Digest (%i updates)" % n
         summary = u"Digest summary:\n"
+        i = 0
         for msg in queued_messages:
+            i = i + 1
             line = fedmsg.meta.msg2subtitle(msg, **self.config) or u''
-            summary += str(i+1) + ".\t" + line + "\n"
+            summary += '%d.\t%s\n' % (i, line)
```
Comment (bowlofeggs) [inline fmn/consumer/backends/mail.py:158]: My suggestion had more been about using `enumerate()` to avoid the `i = 0` and the `i = i + 1`, but you can leave it this way if you like since it's functionally equivalent.

**Review by bowlofeggs — APPROVED** [review]
Approved!

My new comments are optional, so feel free to merge as-is.
